src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -61,6 +63,68 @@ internal partial class RenamerProjectTreeActionHandler : ProjectTreeActionHandle
             _operationProgressService = operationProgressService;
         }
 
+        public override async Task CopyAsync(IProjectTreeActionHandlerContext context, string filename, string folderToMove)
+        {
+            Requires.NotNull(context, nameof(Context));
+            Requires.NotNull(filename, nameof(filename));
+            Requires.NotNull(folderToMove, nameof(folderToMove));
+            // Get actions to update namespace
+            CodeAnalysis.Project? project = GetCurrentProject();
+
+            if (project is null)
+            {
+                return;
+            }
+
+            string folderPath = Path.GetDirectoryName(folderToMove);
+            folderPath = GetLastDirectory(folderPath);
```

Comment:
If you're trying to get the name of the last directory in the path, you can do `Path.GetFileName(Path.GetDirectoryName(path))` and you wouldn't need the local method.

Reply:
done in next commit